src/main/java/org/folio/search/service/setter/instance/IsbnProcessor.java

```diff
@@ -0,0 +1,127 @@
+
+  private final Set<String> isbnIdentifierTypeIds =
+    Set.of(ISBN_IDENTIFIER_TYPE_ID, INVALID_ISBN_IDENTIFIER_TYPE_ID);
+
+  /**
+   * Used by dependency injection.
+   *
+   * @param jsonConverter {@link JsonConverter} bean
+   */
+  public IsbnProcessor(JsonConverter jsonConverter) {
+    super(jsonConverter);
+  }
+
+  @Override
+  public List<String> getFieldValue(Map<String, Object> eventBody) {
+    var isbnIdentifiers = new LinkedHashSet<String>();
+
+    for (var identifier : getInstanceIdentifiers(eventBody)) {
+      var normalizedValue = normalizeIsbn(identifier.getValue());
+      isbnIdentifiers.addAll(normalizedValue);
+    }
+
+    return new ArrayList<>(isbnIdentifiers);
+  }
+
+  @Override
+  protected Set<String> getIdentifierTypeIds() {
+    return isbnIdentifierTypeIds;
+  }
+
+  private static List<String> normalizeIsbn(String value) {
+    String isbnValue = StringUtils.trim(value);
+    if (StringUtils.isEmpty(isbnValue)) {
+      return emptyList();
+    }
+    var isbn13Matcher = ISBN13_REGEX.matcher(isbnValue);
+    if (isbn13Matcher.find() && isValid13DigitNumber(isbn13Matcher.group(0))) {
+      return getNormalizedIsbnValue(isbn13Matcher, singletonList(isbn13Matcher.group(0)));
+    }
+
+    var isbn10Matcher = ISBN10_REGEX.matcher(isbnValue);
+    if (isbn10Matcher.find() && isValid10DigitNumber(isbn10Matcher.group(0))) {
+      var isbn10Value = isbn10Matcher.group(0);
+      return getNormalizedIsbnValue(isbn10Matcher, List.of(isbn10Value, convertTo13DigitNumber(isbn10Value)));
+    }
+
+    return List.of(replaceCharactersBetweenDigits(isbnValue));
+  }
+
+  private static List<String> getNormalizedIsbnValue(Matcher isbnRegexMatcher, List<String> isbnValues) {
+    var normalizedIsbnTokens = new ArrayList<String>();
+    for (String isbnValue : isbnValues) {
+      normalizedIsbnTokens.add(normalizeIsbnValue(isbnValue));
+    }
+    var isbnQualifierValue = isbnRegexMatcher.replaceFirst("").trim();
+    if (StringUtils.isNotBlank(isbnQualifierValue)) {
+      normalizedIsbnTokens.add(isbnQualifierValue);
+    }
+    return normalizedIsbnTokens;
+  }
+
+  /**
+   * Normalizes of invalid isbn value to prevent creation of tokens which can contain partial value.
+   *
+   * @param value value to clean up.
+   * @return string value where all space and hyphen characters between digits are removed
+   */
+  private static String replaceCharactersBetweenDigits(String value) {
+    var resultBuilder = new StringBuilder(value.length());
+    resultBuilder.append(value.charAt(0));
+    for (int i = 1; i < value.length() - 1; i++) {
+      var current = value.charAt(i);
+      if (!(isBetweenNumberCharacters(value.charAt(i - 1), value.charAt(i + 1)) && isHyphenOrSpace(current))) {
+        resultBuilder.append(current);
+      }
+    }
+    resultBuilder.append(value.charAt(value.length() - 1));
+    return resultBuilder.toString();
+  }
```

Comment:
This method will convert `12 3` to `123` but  will not change`12  3` (note the double space in second case). Is this by design?

Reply:
I've implemented cleaning of input string from multiple spaces